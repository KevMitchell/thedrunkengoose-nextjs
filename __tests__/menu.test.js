import { render, fireEvent } from '@testing-library/react'
import Menu from '../pages/menu'

jest.mock('../public/data/drinks.json', () => ([
  {
    name: 'Test Drink1',
    image: 'test.jpg',
    ingredients: [
      { name: 'test ingredient 1', quantity: 10 },
      { name: 'test ingredient 2', quantity: 20 }
    ]
  },
  {
    name: 'Test Drink2',
    image: 'test2.jpg',
    ingredients: [
      { name: 'test ingredient 1', quantity: 10 },
      { name: 'test ingredient 3', quantity: 30 }
    ]
  },
  {
    name: 'Virgin drink',
    image: 'virgin.jpg',
    virgin: true,
    ingredients: [
      { name: 'virgin ingredient 1', quantity: 10 },
      { name: 'virgin ingredient 3', quantity: 30 }
    ]
  }
]))

jest.mock('../public/data/instructions.json', () => ({
  'Test Drink1': [
    'Instruction 1',
    'Instruction 2'
  ]
}))

test('the menu displays drinks and selection items', () => {
  const { container } = render(<Menu />)
  expect(container).toMatchSnapshot()
})

test('the menu shows alcoholic drinks by default', () => {
  const { queryByText } = render(<Menu />)
  const alcoholicDrink1 = queryByText(/Test Drink1/)
  const alcoholicDrink2 = queryByText(/Test Drink2/)
  const virginDrink = queryByText(/Virgin drink/)
  expect(alcoholicDrink1).toBeInTheDocument()
  expect(alcoholicDrink2).toBeInTheDocument()
  expect(virginDrink).toBeNull()
})

test('the menu shows virgin drinks when virgin is selected', () => {
  const { queryByText } = render(<Menu />)
  const virginButton = queryByText(/Virgin/)
  virginButton.click()

  const alcoholicDrink1 = queryByText(/Test Drink1/)
  const alcoholicDrink2 = queryByText(/Test Drink2/)
  const virginDrink = queryByText(/Virgin drink/)
  expect(alcoholicDrink1).toBeNull()
  expect(alcoholicDrink2).toBeNull()
  expect(virginDrink).toBeInTheDocument()
})

test('the menu shows the instructions panel for a clicked on drink', () => {
  const { getByText, getByTestId } = render(<Menu />)
  const drinkButton = getByText(/Test Drink1/)
  drinkButton.click()

  expect(getByTestId('instructionPanel')).toHaveStyle('opacity: 1')
  expect(getByText(/Instruction 1/)).toBeInTheDocument()
  expect(getByText(/Instruction 2/)).toBeInTheDocument()
})

test('the instructions panel is removed when the escape key is pressed', () => {
  const { getByText, getByTestId } = render(<Menu />)
  const drinkButton = getByText(/Test Drink1/)
  drinkButton.click()
  fireEvent.keyDown(document, { key: 'Escape', code: 'Escape' })

  expect(getByTestId('instructionPanel')).toHaveStyle('opacity: 0')
})
