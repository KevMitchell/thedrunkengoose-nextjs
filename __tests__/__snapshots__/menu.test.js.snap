// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the menu displays drinks and selection items 1`] = `
<div>
  <div
    class="jsx-2657862061"
  >
    <span
      class="jsx-2657862061 selected"
    >
      Alcoholic
    </span>
    <span
      class="jsx-2657862061 "
    >
      Virgin
    </span>
  </div>
  <div
    class="jsx-2656356009 instructionPanel"
    data-testid="instructionPanel"
  >
    <div
      class="jsx-2656356009 title"
    >
      Instructions
    </div>
    <div
      class="jsx-2656356009 close"
    >
      X
    </div>
    <div
      class="jsx-2656356009 titleBar"
    >
      <div
        class="jsx-2656356009 titleBarText"
      >
        <div
          class="jsx-2656356009 name"
        />
        <div
          class="jsx-2656356009 ingredients"
        />
      </div>
    </div>
    <div
      class="jsx-2656356009 instructions"
    />
  </div>
  <div
    class="jsx-1499698495 drinkList"
  >
    <div
      class="jsx-2343412649 drinkItem"
    >
      <div
        class="jsx-2343412649 name"
      >
        Test Drink1
      </div>
      <div
        class="jsx-2343412649 imageContainer"
      >
        <img
          alt="Test Drink1"
          class="jsx-2343412649 image"
          src="./img/drinks/test.jpg"
        />
      </div>
      <div
        class="jsx-2343412649 ingredients"
      >
        <div>
          test ingredient 1
        </div>
        <div>
          test ingredient 2
        </div>
      </div>
    </div>
    <div
      class="jsx-2343412649 drinkItem"
    >
      <div
        class="jsx-2343412649 name"
      >
        Test Drink2
      </div>
      <div
        class="jsx-2343412649 imageContainer"
      >
        <img
          alt="Test Drink2"
          class="jsx-2343412649 image"
          src="./img/drinks/test2.jpg"
        />
      </div>
      <div
        class="jsx-2343412649 ingredients"
      >
        <div>
          test ingredient 1
        </div>
        <div>
          test ingredient 3
        </div>
      </div>
    </div>
  </div>
</div>
`;
